Update gradle version for AndroidNative app from 7.6.1 to 8.10.2.

LoopedBard3 opened the pull request:
### Description of Change

Updates the gradle version for the AndroidNative app from 7.6.1 to 8.12.1. Once backported to net9.0 this will fix a performance pipeline issue that is being hit due to the gradle version being too old, specifically: `BUG! exception in phase 'semantic analysis' in source unit '_BuildScript_' Unsupported class file major version 65` when gradle is running during the build of the app at `src/Core/tests/Benchmarks.Droid/Benchmarks.Droid.csproj`.

This also needs to be backported to the net9.0 branch.

LoopedBard3 committed:
Update gradle version for AndroidNative app from 7.6.1 to 8.12.1.

jsuarezruiz commented:
/azp run

PureWeen reviewed (changes requested):
failing to build

/Users/builder/azdo/_work/3/s/.dotnet/packs/Microsoft.Android.Sdk.Darwin/35.0.39/tools/Microsoft.Android.Sdk.Bindings.Gradle.targets(83,5): error XAGRDL0000:    > Namespace not specified. Specify a namespace in the module's build file: /Users/builder/azdo/_work/3/s/src/Core/AndroidNative/maui/build.gradle. See https://d.android.com/r/tools/upgrade-assistant/set-namespace for information about setting the namespace. [/Users/builder/azdo/_work/3/s/src/Core/src/Core.csproj::TargetFramework=net9.0-android35.0]
/Users/builder/azdo/_work/3/s/.dotnet/packs/Microsoft.Android.Sdk.Darwin/35.0.39/tools/Microsoft.Android.Sdk.Bindings.Gradle.targets(83,5): error XAGRDL0000:       [/Users/builder/azdo/_work/3/s/src/Core/src/Core.csproj::TargetFramework=net9.0-android35.0]

jonathanpeppers committed:
Use Android Studio to do the gradle upgrade

It selected `gradle-8.10.2` but then also updated other files.

Let's give this a try.

jonathanpeppers commented:
/azp run

jonathanpeppers committed:
Set `namespace`

> A problem occurred configuring project ':maui'.
> Could not create an instance of type com.android.build.api.variant.impl.LibraryVariantBuilderImpl.
   > Namespace not specified. Specify a namespace in the module's build file: /Users/builder/azdo/_work/3/s/src/Core/AndroidNative/maui/build.gradle. See https://d.android.com/r/tools/upgrade-assistant/set-namespace for information about setting the namespace.

jonathanpeppers commented:
/azp run

jonathanpeppers committed:
Delete AndroidManifest.xml

package="com.microsoft.maui" found in source AndroidManifest.xml
Setting the namespace via the package attribute in the source AndroidManifest.xml is no longer supported, and the value is ignored.
Recommendation: remove package="com.microsoft.maui" from the source AndroidManifest.xml:

jonathanpeppers committed:
Update PlatformInterop.java

TintTypedArray styledAttributes = TintTypedArray.obtainStyledAttributes(context, null, R.styleable.Toolbar, resId, 0);

jonathanpeppers commented:
/azp run

PureWeen commented:
/rebase

PureWeen commented:
https://github.com/dotnet/maui/pull/30543
